File: Random/04-random-30/Search Insert Position.go
package main

func searchInsert(nums []int, target int) int {
	l := -1
	r := len(nums) - 1
	for r-l > 1 {
		m := (l + r) >> 1
		if nums[m] >= target {
			r = m
		} else {
			l = m
		}
	}
	if nums[r] < target {
		r++
	}
	return r
}

File: Random/04-random-30/Search Insert Position.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution
{
public:
    int searchInsert(vector<int> &nums, int target)
    {
        int l = 0;
        int r = nums.size() - 1;
        int mid;
        while (l <= r)
        {
            mid = (l + r) / 2;
            if (nums[mid] == target)
            {
                return mid;
            }
            else if (nums[mid] < target)
            {
                l = mid + 1;
            }
            else
                r = mid - 1;
        }
        return l;
    }
};